front/lib/api/data_sources.ts

```diff
@@ -1280,12 +1281,38 @@ export async function unpauseAllManagedDataSources(
 }
 
 export async function computeDataSourceStatistics(
-  dataSource: DataSourceResource
+  dataSource: DataSourceResource[]
 ) {
   const coreAPI = new CoreAPI(config.getCoreAPIConfig(), logger);
 
   return coreAPI.getDataSourceStats({
-    projectId: dataSource.dustAPIProjectId,
-    dataSourceId: dataSource.dustAPIDataSourceId,
+    projectIds: dataSource.map(({ dustAPIProjectId }) => dustAPIProjectId),
+    dataSourceIds: dataSource.map(
+      ({ dustAPIDataSourceId }) => dustAPIDataSourceId
+    ),
   });
 }
+
+export const computeWorkspaceOverallSizeCached = cacheWithRedis(
+  async (auth: Authenticator) => {
+    const dataSources = await DataSourceResource.listByWorkspace(
+      auth,
+      // TODO(DATASOURCE_SID): Clean-up
+      { origin: "v1_data_sources_documents_document_get_or_upsert" }
+    );
+    const result = await computeDataSourceStatistics(dataSources);
+
+    if (result.isErr()) {
+      throw new Error(
+        `Failed to get data source stats: ${result.error.message}`
+      );
+    }
+
+    return result.value.overall_total_size;
+  },
+  (auth: Authenticator) => {
+    const workspaceId = auth.getNonNullableWorkspace().sId;
+    return `compute-datasource-stats:${workspaceId}`;
+  },
+  60 * 10 * 1000 // 10 minutes
```

Comment:
According to our discussions, it seemed that we were open to increase the delay to an hour, correct?

Reply:
:+1: